test/baseVisitors.test.ts

```diff
@@ -0,0 +1,333 @@
+  { type: 'ThisExpression', code: 'this', mock: mock.method(baseVisitor, 'ThisExpression') },
+  { type: 'ThrowStatement', code: 'throw 1', mock: mock.method(baseVisitor, 'ThrowStatement') },
+  { type: 'TryStatement', code: 'try {} catch(e) {}', mock: mock.method(baseVisitor, 'TryStatement') },
+  { type: 'TsArrayType', code: 'type a = string[]', mock: mock.method(baseVisitor, 'TsArrayType'), debug: true },
+  //   TsExpressionWithTypeArguments<S>(n: swc.TsExpressionWithTypeArguments, st: S, cb: Callback<S>) {
+  //     cb(n.expression, st)
+
+  //     if (n.typeArguments) {
+  //       cb(n.typeArguments, st)
+  //     }
+  //   }
+  //   TsInterfaceDeclaration<S>(n: swc.TsInterfaceDeclaration, st: S, cb: Callback<S>) {
+  //     cb(n.id, st)
+  //     cb(n.body, st)
+
+  //     for (const ext of n.extends) {
+  //       cb(ext, st)
+  //     }
+
+  //     if (n.typeParams) {
+  //       cb(n.typeParams, st)
+  //     }
+  //   }
+  //   TsInterfaceBody<S>(n: swc.TsInterfaceBody, st: S, cb: Callback<S>) {
+  //     for (const ele of n.body) {
+  //       cb(ele, st)
+  //     }
+  //   }
+  //   TsKeywordType = ignore
+  //   TsPropertySignature<S>(n: swc.TsPropertySignature, st: S, cb: Callback<S>) {
+  //     for (const param of n.params) {
+  //       cb(param, st)
+  //     }
+
+  //     cb(n.key, st)
+
+  //     if (n.typeAnnotation) {
+  //       cb(n.typeAnnotation, st)
+  //     }
+
+  //     if (n.typeParams) {
+  //       cb(n.typeParams, st)
+  //     }
+  //   }
+  //   TsAsExpression<S>(n: swc.TsAsExpression, st: S, cb: Callback<S>) {
+  //     cb(n.expression, st)
+  //     cb(n.typeAnnotation, st)
+  //   }
+  //   TsCallSignatureDeclaration<S>(n: swc.TsCallSignatureDeclaration, st: S, cb: Callback<S>) {
+  //     for (const param of n.params) {
+  //       cb(param, st)
+  //     }
+  //     if (n.typeParams) {
+  //       cb(n.typeParams, st)
+  //     }
+  //     if (n.typeAnnotation) {
+  //       cb(n.typeAnnotation, st)
+  //     }
+  //   }
+  {
+    type: 'TsExpressionWithTypeArguments',
+    code: 'type foo<T> = Array<T>',
+    mock: mock.method(baseVisitor, 'TsExpressionWithTypeArguments'),
+    skip: true,
+  },
+  {
+    type: 'TsInterfaceDeclaration',
+    code: 'interface Foo {}',
+    mock: mock.method(baseVisitor, 'TsInterfaceDeclaration'),
+  },
+  { type: 'TsInterfaceBody', code: 'interface Foo {}', mock: mock.method(baseVisitor, 'TsInterfaceBody') },
+  { type: 'TsKeywordType', code: 'type foo = string', mock: mock.method(baseVisitor, 'TsKeywordType') },
+  {
+    type: 'TsPropertySignature',
+    code: 'type foo = {a: string}',
+    mock: mock.method(baseVisitor, 'TsPropertySignature'),
+  },
+  { type: 'TsAsExpression', code: 'a as b', mock: mock.method(baseVisitor, 'TsAsExpression') },
+  {
+    type: 'TsCallSignatureDeclaration',
+    code: 'type foo = () => void',
+    mock: mock.method(baseVisitor, 'TsCallSignatureDeclaration'),
```

Comment:
Not entirely sure why yet, but when I ran the tests locally this one failed:

```
✖ should call TsCallSignatureDeclaration (157.277577ms)
  AssertionError [ERR_ASSERTION]: Expected TsCallSignatureDeclaration to be called 1 times, but was called 0 times.
      at Object.toHaveBeenCalledTimes (/Users/money.ctr/morganney/swc-walk/test/baseVisitors.test.ts:326:14)
      at TestContext.<anonymous> (/Users/money.ctr/morganney/swc-walk/test/baseVisitors.test.ts:310:28)
      at Test.runInAsyncScope (node:async_hooks:206:9)
      at Test.run (node:internal/test_runner/test:639:25)
      at Test.processPendingSubtests (node:internal/test_runner/test:382:18)
      at Test.postRun (node:internal/test_runner/test:730:19)
      at Test.run (node:internal/test_runner/test:688:12)
      at async Test.processPendingSubtests (node:internal/test_runner/test:382:7) {
    generatedMessage: false,
    code: 'ERR_ASSERTION',
    actual: 0,
    expected: 1,
    operator: '=='
  }
```

Reply:
The problem with the AST node is to find, the correct parents nodes
for the test whose are ready yet, I've either relied on Github Copilot or astexplorer.net